contracts/CTTreasury.sol

```diff
@@ -266,10 +268,16 @@ contract CTTreasury is SolidDaoManaged, ERC1155Receiver {
         uint256 _amount,
         address _owner
     ) external returns (bool) {
-        require(permissions[STATUS.RESERVETOKEN][_token], "SCT Treasury: reserve token not permitted");
-        require(carbonProjects[_token][_tokenId].isActive, "SCT Treasury: carbon project not active");
-        require((IERC1155(_token).balanceOf(_owner, _tokenId)) >= _amount, "SCT Treasury: owner insuficient ERC1155 balance");
-        require((IERC1155(_token).isApprovedForAll(_owner, address(this))) , "SCT Treasury: owner not approved this contract spend ERC1155");
+        require(initialized, "Contract was not initialized yet");
+        require(permissions[STATUS.RESERVETOKEN][_token], "CT Treasury: reserve token not permitted");
+        require(carbonProjects[_token][_tokenId].isActive, "CT Treasury: carbon project not active");
+        require((IERC1155(_token).balanceOf(_owner, _tokenId)) >= _amount, "CT Treasury: owner insuficient ERC1155 balance");
+        require((IERC1155(_token).isApprovedForAll(_owner, address(this))) , "CT Treasury: owner not approved this contract spend ERC1155");
+        
+        (bool mathOK, uint256 weeksFromDelivery) = SolidMath.weeksInThePeriod(block.timestamp, carbonProjects[_token][_tokenId].contractExpectedDueDate);
```

Comment:
nit: Rename `weeksFromDelivery` to `weeksToDelivery`

Reply:
adjusted